tests/drivers/can/api/src/main.c

```diff
@@ -1122,42 +1163,49 @@ static void test_filters_preserved_through_bitrate_change(void)
 	can_remove_rx_filter(can_dev, filter_id);
 }
 
-void test_main(void)
+/**
+ * @brief Test that the prequisites for the test suite are met.
+ */
+ZTEST(can_api, test_prerequisites)
+{
+	zassert_true(device_is_ready(can_dev), "CAN device not ready");
+	zassert_equal(can_dev_init_status, 0, "failed to set loopback mode (err %d)",
+		      can_dev_init_status);
```

Comment:
If I could just do these asserts in the `can_api_setup()` function and have the entire test suite error-out (not skip, but report an error/failure)...

Reply:
So the proposed method here is:

Have a single tiny test suite like `ZTEST_SUITE(can_api_setup)` that only tests the setup success. After that, the `ZTEST_SUITE(can_api)` suite can safely assume the in the setup/before because if the assumption fails and skips the suite, we'll see the failure in the `can_api_setup` suite.

Reply:
Which suite would call the setup function? Only one of them can call it and the suite execution order is not guaranteed.

Reply:
tldr: 
Yeah, this is more difficult than it should be. What you have right now works,
but it should be easier than this.

Ahh okay. So I decided to pull down the changes and try to make them myself and
I definitely get it now.

In my opinion, this PR is a great case-example for zasserts being supported in
the setup since we can only call `can_set_mode()` once and only once.

My original idea was to have two separate ZTEST_SUITES, one small one with a
single test that verifies the `can_set_mode()` and another that assumes
`can_set_mode()` and then goes along with the other tests. But as mentioned,
`can_set_mode()` can only be called once.

So the `predicate` parameter in the ZTEST_SUITE would be an existing solution to
this problem. Where we run two suites (one suite testing prequisites another
testing other requirements) and selectively choose one suite at a time. But I
don't believe it's intended for such a simple example as this. Let me know if
you'd like me to show you the change here. 

The alternative would be running each one in its own binary, but that seems a
bit much to me as well. I'll raise this and get a lot more attention on this
Monday. Thank you! I really appreciate you finding this example and notifying me
about it.

Would you like me to approve your PR as it stands now?

Reply:
> Would you like me to approve your PR as it stands now?

Thanks, but no. I'd rather wait for @yperess' work on being able to use `zassert` in `setup` and convert all the CAN drivers tests in one go.